python/acquire/__init__.py

```diff
@@ -167,31 +167,23 @@ def is_not_done() -> bool:
                 nframes[1] < p.video[1].max_frame_count
             )
 
-        while is_not_done():  # runtime.get_state()==DeviceState.Running:
-            clock = time.time()
-
+        def next_frame() -> Optional[npt.NDArray[Any]]:
+            """Get the next frame from the current stream."""
             if nframes[stream_id] < p.video[stream_id].max_frame_count:
-
                 if packet := runtime.get_available_data(stream_id):
                     n = packet.get_frame_count()
-                    f = next(packet.frames())
-                    im = f.data().squeeze().copy()
-                    logging.debug(
-                        f"stream {stream_id} frame {f.metadata().frame_id}"
-                    )
-
-                    # TODO: (nclack) fix this awkwardness.
```

Comment:
I think this should be fixed by extracting this code block into a function, since variables in Python are functionally scoped.